components/join_private_channel_modal.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import ConfirmModal from 'components/confirm_modal';
+
+type Props = {
+    channelName: string;
+    onCancel: () => void;
+    onHide: () => void;
+    onJoin: () => void;
+}
+
+type State = {
+    show: boolean;
+}
+
+export default class JoinPrivateChannelModal extends React.PureComponent<Props, State> {
```

Comment:
Should we make this into a functional component? It doesn't have any lifecycle functions and only takes a few props

Reply:
Yes .. I wasn't thinking about that. Good one!